cinder/autogenerated_client.go

```diff
@@ -0,0 +1,1821 @@
+package cinder
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+type RequestHandlerFn func(*http.Request) (*http.Response, error)
+
+type UpdateVolumeTypeParams struct {
+
+	// VolumeType is required.
+	//
+	// A volume type offers a way to categorize or group volumes.
+	VolumeType string `json:"volume_type"`
+
+	//
+	// The unique identifier of the tenant or account.
+	TenantId string `json:"-"`
+
+	// VolumeTypeId is required.
+	//
+	// The unique identifier for an existing volume type.
+	VolumeTypeId string `json:"-"`
+}
+
+type UpdateVolumeTypeResults struct {
+	VolumeType struct {
+		ExtraSpecs struct {
+			Capabilities string `json:"capabilities"`
+		} `json:"extra_specs"`
+		ID   string `json:"id"`
+		Name string `json:"name"`
+	} `json:"volume_type"`
+}
+
+//
+// Updates a volume type.
+func updateVolumeType(request RequestHandlerFn, args UpdateVolumeTypeParams) (*UpdateVolumeTypeResults, error) {
```

Comment:
@axw: Originally I did have these exported. I agree that it would probably be helpful to give the user flexibility in how they interact with the API.

@wallyworld: These are the same types that are utilized in the public functions, and they will always represent the information the user must specify in order to make a request. I agree that we should offer a curated experience, but I think I disagree that we should try and hide the raw calls from developers. If we make a breaking change, we can always use a version bump.

I am open to this idea. Can you give me an example of where this might be problematic to help clarify?